lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1083,6 +1083,7 @@ def reset_stat(tarinfo):
                         tarinfo.mode = 0o0755 if existing_is_exec or tarinfo.isdir() else 0o0644
                     tarinfo.uid = tarinfo.gid = 0
                     tarinfo.uname = tarinfo.gname = ''
+                    tarinfo.mtime = time.time()
```

Comment:
reset_stat is also used when dealing with normal files
```python
# Dealing with a normal file, just add it by name.
tar_file.add(os.path.realpath(b_src_path), arcname=filename, recursive=False, filter=reset_stat)
```
We probably don't want to overwrite the original mtime in that case.